inference-engine/tests/functional/plugin/shared/src/subgraph_tests/concat_split_relu.cpp

```diff
@@ -0,0 +1,85 @@
+        auto mem_r = std::make_shared<ngraph::opset3::ReadValue>(mem_c, "id");
+
+        auto cnc = std::make_shared<ngraph::op::Concat>(ngraph::OutputVector{input[0], mem_r}, 1);
+
+        auto split1 = ngraph::builder::makeSplit(cnc, ngPrc, 2, 1);
+        auto split2 = ngraph::builder::makeSplit(cnc, ngPrc, 2, 1);
+        auto split3 = ngraph::builder::makeSplit(cnc, ngPrc, 2, 1);
+        auto split4 = ngraph::builder::makeSplit(cnc, ngPrc, 2, 1);
+
+        auto reshape_pattern = std::make_shared<ngraph::op::Constant>(ngraph::element::i64, ngraph::Shape{3}, std::vector<size_t>{1, 128, 2});
+        auto permute_pattern = std::make_shared<ngraph::op::Constant>(ngraph::element::i64, ngraph::Shape{3}, std::vector<size_t>{0, 2, 1});
+
+        auto reshape1 = std::make_shared<ngraph::opset1::Reshape>(split2->output(1), reshape_pattern, false);
+        auto reshape2 = std::make_shared<ngraph::opset1::Reshape>(split3->output(1), reshape_pattern, false);
+        auto reshape3 = std::make_shared<ngraph::opset1::Reshape>(split4->output(1), reshape_pattern, false);
+
+        auto permute1 = std::make_shared<ngraph::opset1::Transpose>(reshape1, permute_pattern);
+        auto permute2 = std::make_shared<ngraph::opset1::Transpose>(reshape2, permute_pattern);
+        auto permute3 = std::make_shared<ngraph::opset1::Transpose>(reshape3, permute_pattern);
+
+        auto reshape_pattern_2 = std::make_shared<ngraph::op::Constant>(ngraph::element::i64, ngraph::Shape{2}, std::vector<size_t>{1, 256});
+        auto reshape_4 = std::make_shared<ngraph::opset1::Reshape>(permute1, reshape_pattern_2, false);
+        auto reshape_5 = std::make_shared<ngraph::opset1::Reshape>(permute2, reshape_pattern_2, false);
+        auto reshape_6 = std::make_shared<ngraph::opset1::Reshape>(permute3, reshape_pattern_2, false);
+
+        auto relu1 = std::make_shared<ngraph::opset1::Relu>(reshape_4);
+        auto relu2 = std::make_shared<ngraph::opset1::Relu>(reshape_5);
+        auto relu3 = std::make_shared<ngraph::opset1::Relu>(reshape_6);
+
+        auto mem_w = std::make_shared<ngraph::opset3::Assign>(split1->output(1), "id");
+
+        ngraph::ResultVector result{std::make_shared<ngraph::opset1::Result>(relu1),
+                                    std::make_shared<ngraph::opset1::Result>(relu2),
+                                    std::make_shared<ngraph::opset1::Result>(relu3),
+                };
+        mem_w->add_control_dependency(mem_r);
+        relu2->add_control_dependency(mem_w);
+
+        function = std::make_shared<ngraph::Function>(result, input, "concat_split_relu");
+    }
+
+    TEST_P(ConcatSplitRelu, CompareWithRefs) {
+        auto ie = InferenceEngine::Core();
+        cnnNetwork = InferenceEngine::CNNNetwork{function};
+
+        auto exe_net = ie.LoadNetwork(cnnNetwork, targetDevice);
+        auto inf_reg = exe_net.CreateInferRequest();
+        ASSERT_NO_THROW(inf_reg.Infer());
```

Comment:
Does this test validate results? Why isn't just Run() function used?

Reply:
Also is this test failing before introducing these changes?